packages/typespec-ts/test/unit/parametersGenerator.spec.ts

```diff
@@ -3,92 +3,153 @@ import { emitParameterFromCadl } from "./util/emitUtil.js";
+            `
+        );
+        assert.ok(parameters);
+        assertEqualContent(
+          parameters?.content!,
+          `
           import { RequestParameters } from "@azure-rest/core-client";
           
           export interface TestQueryParamProperties {
-            "api-version": string;
+              "user-custom-query": string;
           }
           
           export interface TestQueryParam {
-            queryParameters: TestQueryParamProperties;
+              queryParameters: TestQueryParamProperties;
           }
           
           export type TestParameters = TestQueryParam & RequestParameters;
           `
-      );
+        );
+      });
+
+      it("should generate offsetDateTime as Date | string", async () => {
+        const parameters = await emitParameterFromCadl(
+          `
+          model QueryParameter {
+            @query
+            executionTo?: offsetDateTime;
+          }
+          op test(...QueryParameter): string;
+          `
+        );
+        assert.ok(parameters);
+        assertEqualContent(
+          parameters?.content!,
+          `
+            import { RequestParameters } from "@azure-rest/core-client";
+            
+            export interface TestQueryParamProperties {
+              executionTo?: Date | string;
+            }
+  
+            export interface TestQueryParam {
+              queryParameters?: TestQueryParamProperties;
+            }
+            
+            export type TestParameters = TestQueryParam & RequestParameters;
+            `
+        );
+      });
     });
   });
 
-  describe("query parameters generation", () => {
-    it("should generate user-custom-query ", async () => {
+  describe("header parameters", () => {
+    it("should generate offsetDateTime as string", async () => {
       const parameters = await emitParameterFromCadl(
         `
-          model CustomParameter {
-            @query
-            "user-custom-query": string;
-          }
-          op test(...CustomParameter): string;
-          `
+        model QueryParameter {
+          @header
+          executionTo?: offsetDateTime;
+        }
+        op test(...QueryParameter): string;
+        `
       );
       assert.ok(parameters);
       assertEqualContent(
         parameters?.content!,
-        `
-        import { RequestParameters } from "@azure-rest/core-client";
-        
-        export interface TestQueryParamProperties {
-            "user-custom-query": string;
-        }
-        
-        export interface TestQueryParam {
-            queryParameters: TestQueryParamProperties;
-        }
-        
-        export type TestParameters = TestQueryParam & RequestParameters;
-        `
+        ` import { RawHttpHeadersInput } from "@azure/core-rest-pipeline";
+          import { RequestParameters } from "@azure-rest/core-client";
+          
+          export interface TestHeaders {
+            "execution-to"?: string;
```

Comment:
why is this different with what has been defined in the typespec ?

Reply:
This handled by tsp compiler not us.

Reply:
@timotheeguerin Hi, do you have any context on this ? The only document I can found about header is that they are case insensitive, but it doesn't say this dash mark can also be ignored. https://datatracker.ietf.org/doc/html/rfc7230#section-3.2

Reply:
The header decorator automatically convert camelCase to camel-case as the header value if you don’t specify an explicit value

Reply:
Is this documented somewhere? I mean in TypeSpec documentation.